sources/packages/backend/libs/integrations/src/services/disbursement-schedule/disbursement-schedule.service.ts

```diff
@@ -199,4 +200,68 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
     const repository = disbursementScheduleRepo ?? this.repo;
     return repository.update({ id: In(disbursementIds) }, { dateSent });
   }
+
+  /**
+   * Fetch the eligible COEs for a given date.
+   * @param generatedDate Date in which the eligible COE is required for an institution.
+   * @returns eligible COEs.
+   */
+  async getEligibleCOE(
+    generatedDate?: string,
+  ): Promise<DisbursementSchedule[]> {
+    const processingDate = generatedDate
+      ? new Date(generatedDate)
+      : addDays(-1);
+    return this.repo.find({
+      select: {
+        id: true,
+        disbursementDate: true,
+        disbursementValues: { id: true, valueCode: true, valueAmount: true },
+        studentAssessment: {
+          id: true,
+          offering: {
+            id: true,
+            studyStartDate: true,
+            studyEndDate: true,
+            institutionLocation: {
+              institutionCode: true,
+            },
+          },
+          application: {
+            id: true,
+            applicationNumber: true,
+            studentNumber: true,
+            student: {
+              id: true,
+              birthDate: true,
+              sinValidation: { id: true, sin: true },
+              user: { id: true, lastName: true, firstName: true },
+            },
+          },
+        },
+      },
+      relations: {
+        disbursementValues: true,
+        studentAssessment: {
+          offering: { institutionLocation: true },
+          application: {
+            student: {
+              sinValidation: true,
+              user: true,
+            },
+          },
+        },
+      },
+      where: {
+        coeStatus: COEStatus.required,
```

Comment:
so here we will be considering coe with the status `required` and date `between date and date +1` right?